Content.Shared/Nutrition/EntitySystems/ThirstSystem.cs

```diff
@@ -154,22 +155,23 @@ private void UpdateEffects(EntityUid uid, ThirstComponent component)
 
     public override void Update(float frameTime)
     {
-        _accumulatedFrameTime += frameTime;
+        base.Update(frameTime);
 
-        if (_accumulatedFrameTime > 1)
+        var query = EntityQueryEnumerator<ThirstComponent>();
+        while (query.MoveNext(out var uid, out var thirst))
         {
-            var query = EntityManager.EntityQueryEnumerator<ThirstComponent>();
-            while (query.MoveNext(out var uid, out var comp))
+            if (_timing.CurTime < thirst.NextUpdateTime)
+                continue;
+            thirst.NextUpdateTime = _timing.CurTime + thirst.UpdateRate;
+
+            UpdateThirst(thirst, -thirst.ActualDecayRate);
+            var calculatedThirstThreshold = GetThirstThreshold(thirst, thirst.CurrentThirst);
+            if (calculatedThirstThreshold != thirst.CurrentThirstThreshold)
             {
-                UpdateThirst(comp, - comp.ActualDecayRate);
-                var calculatedThirstThreshold = GetThirstThreshold(comp, comp.CurrentThirst);
-                if (calculatedThirstThreshold != comp.CurrentThirstThreshold)
-                {
-                    comp.CurrentThirstThreshold = calculatedThirstThreshold;
-                    UpdateEffects(uid, comp);
-                }
+                thirst.CurrentThirstThreshold = calculatedThirstThreshold;
+                UpdateEffects(uid, thirst);
             }
-            _accumulatedFrameTime -= 1;
+            Dirty(thirst);
```

Comment:
Can't the client just derive the current value from `(current time - last change) * decay rate` rather than dirtying a mob every single second.

Reply:
This only dirties the mob if a threshold has been reached.